plugins/saveactions.c

```diff
@@ -441,6 +969,63 @@ void plugin_init(GeanyData *data)
 	store_target_directory(tmp, &backupcopy_backup_dir);
 	g_free(tmp);
 
+	untitled_doc_default_ft = utils_get_setting_string(config, "untitled_document_save", "default_ft",
+		filetypes[GEANY_FILETYPES_NONE]->name);
+
+	tmp = utils_get_setting_string(config, "untitled_document_save", "instantsave_target_dir", NULL);
+	store_target_directory(tmp, &instantsave_target_dir);
+	g_free(tmp);
+
+	/* START Persistent untitled documents */
+	if (utils_get_setting_string(config, "untitled_document_save", "persistent_untitled_documents_target_dir", NULL) == NULL)
+	{
+		/* Set default target dir */
+		gchar *configdir_utf8, *default_persistent_doc_dir_utf8;
+
+		configdir_utf8 = utils_get_utf8_from_locale(geany->app->configdir);
+		default_persistent_doc_dir_utf8 = g_strconcat(configdir_utf8, G_DIR_SEPARATOR_S, "plugins",
+			G_DIR_SEPARATOR_S, "saveactions", G_DIR_SEPARATOR_S, "persistent_untitled_documents", NULL);
+		g_free(configdir_utf8);
+
+		g_key_file_set_string(config, "untitled_document_save", "persistent_untitled_documents_target_dir",
+			default_persistent_doc_dir_utf8);
+
+		tmp = utils_get_locale_from_utf8(default_persistent_doc_dir_utf8);
+		g_free(default_persistent_doc_dir_utf8);
+
+		utils_mkdir(tmp, TRUE);
+		g_free(tmp);
+	}
+
+	tmp = utils_get_setting_string(config, "untitled_document_save", "persistent_untitled_documents_target_dir", NULL);
+	SETPTR(tmp, utils_get_locale_from_utf8(tmp));
+	/* Set target dir variable with value from config, regardless if dir is valid or not */
+	SETPTR(persistent_docs_target_dir, g_strdup(tmp));
+
+	if (enable_persistent_docs && ! is_directory_accessible(tmp))
+	{
+		/* switch functionality off, so invalid target directory cannot be actually used */
+		enable_persistent_docs = FALSE;
+		g_key_file_set_boolean(config, "saveactions", "enable_persistent_untitled_documents", FALSE);
+
+		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(pref_widgets.untitled_doc_disabled_radio), TRUE);
+		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(pref_widgets.untitled_doc_persistent_radio), FALSE);
```

Comment:
This line should not be necessary I believe because of the previous line and because it's a radio.

Reply:
same - each one launches its own handler which disabled ui elements of unused features - `radio_toggled_cb()`
This code could be written in some other way but existing repeats existing handling for checkboxes and is generally ok